### ISSUE TITLE
[a11y]: TextArea + TextInput High Contrast Focus Indicator Issue

### ISSUE THREAD

**Issue by RachelGHogan** [carbon-design-system/carbon#17653]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.67.0

### Automated testing tool and ruleset

Chrome Devtools High Contrast Mode

### Assistive technology

Chrome Devtools High Contrast Mode

### Description

Currently, the only focus indicator for TextArea and TextInputs is the blinking cursor, which isn't very visible especially for high contrast mode.
![Image](https://github.com/user-attachments/assets/a91c46ac-387a-4c2f-b78d-3387725c83cb)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-textinput--default

### Steps to reproduce

Click the link provided above to go to the Storybook website. Turn Emulate "Emulate CSS media feature forced-colors" on and click or tab into the text box to observe there is no change in border or outline to indicate focus.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by annawen1** [comment]
@Kritvi-bhatia17 @chrisgrimley looks like there is a change in the focus outline, but the color is very faint. Is there an alternative color or approach for this?

https://github.com/user-attachments/assets/337e0401-f127-4d9c-a5f3-d6c6a787a4d1

**Comment by chrisgrimley** [comment]
Hi @annawen1 and @Kritvi-bhatia17. Depending on how you interpret the WCAG guidelines, this may not be in violation. Under SC 2.4.13 there is an exception that reads:

_In combination with [2.4.7 Focus Visible](https://www.w3.org/WAI/WCAG22/Understanding/focus-visible), [1.4.11 Non-text Contrast](https://www.w3.org/WAI/WCAG22/Understanding/non-text-contrast) requires that the visual focus indicator for a component must have sufficient contrast against the adjacent colors when the component is focused, **except where the appearance of the component is determined by the user agent and not modified by the author**._

However, it also says that the exception only applies if "the focus indicator cannot be adjusted by the author" and from what I know this can be easily addressed by having a CSS 'forced-colors' fallback. It might be worth discussing this update with the engineers to see if this can be added. 

Let me know if you need anything more from me.

**Comment by annawen1** [comment]
https://github.com/user-attachments/assets/08cbf3e5-310e-4adf-9043-5bccb7ccdb42

@Kritvi-bhatia17 we do have a `high-contrast-mode` mixin that we can add to the text-input component. Currently if I apply the mixin as is, it makes the outline 1px on focus (shown in video above). Maybe we can increase the focus outline width to 3px when user has `forced-colors` set?

https://github.com/user-attachments/assets/d565cacb-fbac-4730-8f24-c20ec45068e4

**Comment by annawen1** [comment]
The fix for this issue could potentially fix the first point of this ticket here around the differentiation between focus and error states: https://github.com/carbon-design-system/carbon/issues/11650

**Comment by alina-jacob** [comment]
The solution explained [here](https://github.com/carbon-design-system/carbon/issues/17653#issuecomment-5093844510) is good to go! 🚀
